### PR TITLE
Agent execution error in json format

### PR DESCRIPTION
**Pull request by jngz-es**
### Description
Return agent execution error in xcontent format. If xcontent building failed, use plain text.

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by lezzago — APPROVED** [review]

**Review by Zhangxunmt — APPROVED** [review]

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLExecuteActionTests.java [hunk 1]
```diff
@@ -281,4 +282,59 @@ public void testPrepareRequestSystemException() throws Exception {
...
+
+        ArgumentCaptor<MLExecuteTaskRequest> argumentCaptor = ArgumentCaptor.forClass(MLExecuteTaskRequest.class);
+        verify(client, times(1)).execute(eq(MLExecuteTaskAction.INSTANCE), argumentCaptor.capture(), any());
+        Input input = argumentCaptor.getValue().getInput();
+        assertEquals(FunctionName.AGENT, input.getFunctionName());
+        ArgumentCaptor<RestResponse> restResponseArgumentCaptor = ArgumentCaptor.forClass(RestResponse.class);
+        verify(channel, times(1)).sendResponse(restResponseArgumentCaptor.capture());
+        BytesRestResponse response = (BytesRestResponse) restResponseArgumentCaptor.getValue();
+        assertEquals(RestStatus.BAD_REQUEST, response.status());
+        assertEquals("text/plain; charset=UTF-8", response.contentType());
```
Comment (dhrubo-os) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLExecuteActionTests.java:335]: Difference between these two tests are only `assertEquals("application/json; charset=UTF-8", response.contentType());` vs `assertEquals("text/plain; charset=UTF-8", response.contentType());`. Is that expected?

Comment (jngz-es) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLExecuteActionTests.java:335]: Yes, it is expected. Actually there is another difference `when(channel.newBuilder()).thenReturn(XContentFactory.jsonBuilder());`
Without this jsonBuilder, the xcontent building will fail, then downgrade to catch branch to send plain text response.

Comment (dhrubo-os) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLExecuteActionTests.java:335]: [nit] I see. In that case, what do you think about something like following:

```
public void testAgentExecutionResponse() throws Exception {
    // Test JSON response
    runAgentExecutionResponseTest(
        "application/json; charset=UTF-8",
        "{\"status\":400,\"error\":{\"type\":\"IllegalArgumentException\",\"reason\":\"Invalid Request\",\"details\":\"Illegal Argument Exception\"}}",
        true
    );

    // Test plain text response
    runAgentExecutionResponseTest(
        "text/plain; charset=UTF-8",
        "{\"error\":{\"reason\":\"Invalid Request\",\"details\":\"Illegal Argument Exception\",\"type\":\"IllegalArgumentException\"},\"status\":400}",
        false
    );
}

private void runAgentExecutionResponseTest(String expectedContentType, String expectedError, boolean isJsonResponse) throws Exception {
    RestRequest request = getExecuteAgentRestRequest();
    
    // Mock client behavior
    doAnswer(invocation -> {
        ActionListener<MLExecuteTaskResponse> actionListener = invocation.getArgument(2);
        actionListener.onFailure(new RemoteTransportException("Remote Transport Exception", new IllegalArgumentException("Illegal Argument Exception")));
        return null;
    }).when(client).execute(eq(MLExecuteTaskAction.INSTANCE), any(), any());
    
    doNothing().when(channel).sendResponse(any());

    // Conditionally mock JSON response building
    if (isJsonResponse) {
        when(channel.newBuilder()).thenReturn(XContentFactory.jsonBuilder());
    }

    restMLExecuteAction.handleRequest(request, channel, client);

    // Verify request processing
    ArgumentCaptor<MLExecuteTaskRequest> argumentCaptor = ArgumentCaptor.forClass(MLExecuteTaskRequest.class);
    verify(client, times(1)).execute(eq(MLExecuteTaskAction.INSTANCE), argumentCaptor.capture(), any());
    Input input = argumentCaptor.getValue().getInput();
    assertEquals(FunctionName.AGENT, input.getFunctionName());

    // Verify response
    ArgumentCaptor<RestResponse> restResponseArgumentCaptor = ArgumentCaptor.forClass(RestResponse.class);
    verify(channel, times(1)).sendResponse(restResponseArgumentCaptor.capture());
    BytesRestResponse response = (BytesRestResponse) restResponseArgumentCaptor.getValue();
    assertEquals(RestStatus.BAD_REQUEST, response.status());
    assertEquals(expectedContentType, response.contentType());
    assertEquals(expectedError, response.content().utf8ToString());
}
```

Comment (jngz-es) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLExecuteActionTests.java:335]: Yeah, this looks good. Actually I would prefer to refactor all tests in `RestMLExecuteActionTests` in next pr, as some existing test cases have the similar improvement to do.

Comment (dhrubo-os) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLExecuteActionTests.java:335]: Sure, approving the PR in expectation of a follow up PR to refactor the tests in `RestMLExecuteActionTests`

**Review by dhrubo-os — APPROVED** [review]

**Pull request merged**